### PR TITLE
BindDnsServer really supports Ubuntu and Debian

### PR DESCRIPTION
**Pull request by sjcorbett**
As the title says, the BindDnsServer entity should work on Debian and Ubuntu machines.  It has been tested on CentOS 6.3, Ubuntu 14.04 and Debian 7.6 VMs. I've also made general improvements to robustness and integration tests.

The easiest way to test is to run Maven with `-PLive -Dtest=BindDnsServerByonLiveTest -DlocationSpec='<a byon spec>'`.

It is worth reviewing the RFC1912 configuration (`rfc1912.zone`) and the three additional zone files: `named.localhost`, `named.loopback` and `named.empty`.

### CODE REVIEW

**Comment by sjcorbett** [comment]
Test failure is a separate issue: `brooklyn.event.feed.jmx.JmxFeedTest.setUp(brooklyn.event.feed.jmx.JmxFeedTest)`

File: software/network/src/main/java/brooklyn/entity/network/bind/BindDnsServerImpl.java [hunk 1]
```diff
@@ -304,6 +275,45 @@ protected void appendTemplate(String template, String destination, SshMachineLoc
                 String.format("rm -f %s", temp)));
     }
 
+    // Mostly used in templates
+
+    protected Predicate<? super Entity> getEntityFilter() {
```
Comment (grkvlt) [inline software/network/src/main/java/brooklyn/entity/network/bind/BindDnsServerImpl.java]: Maybe `public` instead, put in interface?

File: software/network/src/main/java/brooklyn/entity/network/bind/BindDnsServerImpl.java [hunk 2]
```diff
@@ -304,6 +275,45 @@ protected void appendTemplate(String template, String destination, SshMachineLoc
...
+
+    public String getDomainName() {
+        return getConfig(DOMAIN_NAME);
+    }
+
+    /**
+     * @return A serial number guaranteed to be valid for use in a modified domain.zone or reverse.zone file.
+     */
+    public long getSerial() {
+        long next = getAttribute(SERIAL) + 1;
```
Comment (grkvlt) [inline software/network/src/main/java/brooklyn/entity/network/bind/BindDnsServerImpl.java]: perhaps synchronize this with an `Object` mutex?

File: software/network/src/main/java/brooklyn/entity/network/bind/BindDnsServerSshDriver.java [hunk 3]
```diff
@@ -126,4 +143,41 @@ private void copyAsRoot(String template, String destination) {
...
+    }
+
+    public String getRfc1912ZonesFile() {
+        return Os.mergePaths(getOsSupport().getConfigDirectory(), "rfc1912.zone");
+    }
+
+    public BindOsSupport getOsSupport() {
+        BindOsSupport result = osSupport;
+        if (result == null) {
+            synchronized(this) {
```
Comment (grkvlt) [inline software/network/src/main/java/brooklyn/entity/network/bind/BindDnsServerSshDriver.java]: Create a specific `Object` mutex for all synchronization, never `this`

**Comment by grkvlt** [comment]
Looks good @sjcorbett. A few small issues to fix, then should be OK to merge...

**Pull request merged**